apps/examples/err_report_demo/err_report_main.c

```diff
@@ -295,17 +295,35 @@ void error_report_process(int argc, char *argv[])
 		printf("%s", ERR_REPORT_USAGE);
 		return;
 	}
-	if (strcmp(argv[2], "send") == 0) {
+	if (strcmp(argv[2], "test") == 0) {
 		if (argc < 4) {
-			error_report_records(NULL);
+			error_report_test(NULL);
 		} else {
-			error_report_records(argv[3]);
+			error_report_test(argv[3]);
 		}
-	} else if (strcmp(argv[2], "test") == 0) {
+	} else if (strcmp(argv[2], "report_send") == 0) {
 		if (argc < 4) {
-			error_report_test(NULL);
+			error_report_records(NULL);
 		} else {
-			error_report_test(argv[3]);
+			error_report_records(argv[3]);
+		}
+	} else if (strcmp(argv[2], "report_print")) {
+		int nbytes_read;
+		char readbuf[1024];
+		nbytes_read = error_report_data_read(readbuf);
+
+		if (nbytes_read > 0) {
+			unsigned i;
+			error_data_t *readptr = (error_data_t *)readbuf;
+			int nentries = nbytes_read/sizeof(error_data_t);
+			for (i = 0; i < nentries; i++) {
+				printf("| %lu.%06lu | %04x | %d | %d | %04x |\n",
+						readptr->timestamp.tv_sec,
+						readptr->timestamp.tv_usec,
+						readptr->pc_value, readptr->module_id,
+						readptr->error_code, readptr->task_addr);
+				readptr++;
+			}
 		}
```

Comment:
No error is reported here?

Reply:
No error, you are right. This part of the code simply prints out whatever errors have been logged up to this point. The error reporting module at present, does not report itself.